File: Leetcode/C++/11. Container With Most Water/solution.cpp
// https://leetcode.com/problems/container-with-most-water/

#include <algorithm>
#include <vector>

using namespace std;

class Solution
{
public:
    int maxArea(vector<int> &height)
    {
        int left = 0, right = height.size() - 1, maxWater = 0;
        while (left < right)
        {
            maxWater = max(maxWater, max(height[left], height[right]) * (right - left));
            if (height[left] > height[right])
            {
                right -= 1;
            }
            else
            {
                left += 1;
            }
        }
        return maxWater;
    }
};

File: Leetcode/Python/11. Container With Most Water/solution.py
# https://leetcode.com/problems/container-with-most-water/


from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        max_water = 0
        while left < right:
            max_water = max(max_water, min(height[left], height[right]) * (right - left))
            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
        return max_water
